src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final double DEFAULT_RADIUS = 10.0;
+    private static final int NUMBER_OF_FIGURE = 5;
+    private static final double MAXIMUM_VALUE = 50.0;
+    private final Circle defaultFigure = new Circle(Color.WHITE.name()
+            .toLowerCase(), DEFAULT_RADIUS);
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(NUMBER_OF_FIGURE);
+        switch (index) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(), randomSize());

```

Comment:
passing expressions as constructor arguments decreases the overall readability

see [here](https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#%D1%81reating-a-figure-dont-pass-expressions-in-constructor)